src/main/java/io/confluent/connect/hdfs/wal/QFSWAL.java

```diff
@@ -0,0 +1,230 @@
+      this.storage.delete(newLockFile);
+      throw new ConnectException("Lock has been acquired by another process");
+    }
+
+    return newLockFile;
+  }
+
+
+  private void startRenamingTimer() {
+    this.timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        renameLockFile();
+      }
+    }, this.lockRefreshIntervalInMs, this.lockTimeoutInMs);
+  }
+
+  private void renameLockFile() {
+    String newLockFile = getNewLockFile();
+
+    try {
+      this.storage.commit(this.lockFile, newLockFile);
+      this.lockFile = newLockFile;
+    } catch (Exception e) {
+      log.error("Failed to rename the file", e);
+    }
+  }
+
+  private List<SimpleEntry<String, Matcher>> findLockFiles() {
+    List<FileStatus> walFiles = this.storage.list(this.lockDir);
+
+    return walFiles.stream()
+            .filter(fileStatus -> fileStatus.isFile())
+            .map((Function<FileStatus, Object>) fileStatus -> fileStatus.getPath()
+                    .getName())
+            .map(Object::toString)
+            .map(fileName -> new SimpleEntry<String, Matcher>(fileName,
+                                                              pattern.matcher(fileName)))
+            .filter(pair -> pair.getValue().matches())
+            .collect(Collectors.toList());
+  }
+
+  @Override
+  public void acquireLease() throws ConnectException {
+    List<SimpleEntry<String, Matcher>> lockFiles = this.findLockFiles();
+    if (lockFiles.size() == 0) {
+      throw new ConnectException("The lock file is not present in the log dir");
+    }
+
+    if (lockFiles.size() > 1) {
+      throw new ConnectException("More than one lock file reside in the log dir");
+    }
+
+    String filename = lockFiles.get(0).getKey();
+    Matcher matcher = lockFiles.get(0).getValue();
+
+    if (!matcher.group(1).equals(this.uuid.toString())) {
+      log.error(
+              "UUID of the lock file %s for %s-%s topic-partition does not match %s uuid",
+              filename,
+              this.topicPartition.topic(),
+              this.topicPartition.partition(),
+              this.uuid);
+      throw new ConnectException("Lock uuid does not match");
+    }
+
+    if (Long.parseLong(matcher.group(2)) < System.currentTimeMillis() - this.lockTimeoutInMs) {
+      throw new ConnectException(
+              "Lock file has not been renamed for more than the threshold");
+    }
+  }
+
+  @Override
+  public void append(String s, String s1) throws ConnectException {
+    this.acquireLease();
+  }
+
+  @Override
+  public void apply() throws ConnectException {
+  }
+
+  @Override
+  public void truncate() throws ConnectException {
+  }
+
+  @Override
+  public void close() throws ConnectException {
+    this.timer.cancel();
+    this.storage.delete(this.lockFile);
+  }
+
+  @Override
+  public String getLogFile() {
+    return null;
+  }
+
+  @Override
+  public FilePathOffset extractLatestOffset() {
+    return null;
```

Comment:
Maybe comment here or somewhere that we're returning null on purpose so this WAL implementation isn't used for finding the latest offset - just for locking.